app/models/tipline_request.rb

```diff
@@ -19,6 +19,12 @@ def self.request_types
   after_commit :update_elasticsearch_field, on: :update
   after_commit :destroy_elasticsearch_field, on: :destroy
 
+  scope :no_articles_sent, ->(project_media_id) {
+    where(associated_type: 'ProjectMedia', associated_id: project_media_id, smooch_report_received_at: 0,
+      smooch_report_update_received_at: 0, smooch_report_sent_at: 0, smooch_report_correction_sent_at: 0
+    ).where.not(smooch_request_type: %w(relevant_search_result_requests irrelevant_search_result_requests timeout_search_requests))
+  }
```

Comment:
Please try this with the same volume as live just to see how performance behaves and/or if we need to add any new index.

Reply:
I did a benchmark using a live DB dump and test against item with `10732` TiplineRequests

1. number_of_tipline_requests_that_never_received_articles_by_time :  `0.008312`
2. has_tipline_requests_that_never_received_articles: `0.012105`
so I think things looks good

Reply:
Awesome! Thanks Sawy,